e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,323 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFiatTFAuth(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Send non TF token but pay fee in TF token while the TF is paused
+	// EXPECTED: Request fails; TF is paused
+
+	originalAmount := math.OneInt()
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "default", originalAmount, noble, noble)
+	alice := w[0] // 1ustake
+	bob := w[1]   // 1ustake
+
+	mintAmount := 100
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.Minter.KeyName(), "fiat-tokenfactory", "mint", alice.FormattedAddress(), fmt.Sprintf("%duusdc", mintAmount))
+	require.NoError(t, err)
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	sendAmount := 1
+	uusdcFee := 5
+	_, err = val.ExecTx(ctx, alice.KeyName(), "bank", "send", alice.KeyName(), bob.FormattedAddress(), fmt.Sprintf("%dustake", sendAmount), "--fees", fmt.Sprintf("%duusdc", uusdcFee))
+	require.ErrorContains(t, err, "the chain is paused")
+
+	bal, err := noble.GetBalance(ctx, alice.FormattedAddress(), "ustake")
+	require.NoError(t, err)
+	require.Equal(t, originalAmount, bal)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Send non TF token but pay fee in TF token while the sender is blacklisted
+	// EXPECTED: Request fails
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, alice)
+
+	_, err = val.ExecTx(ctx, alice.KeyName(), "bank", "send", alice.KeyName(), bob.FormattedAddress(), fmt.Sprintf("%dustake", sendAmount), "--fees", fmt.Sprintf("%duusdc", uusdcFee))
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not send tokens: unauthorized", alice.FormattedAddress()))
+
+	bal, err = noble.GetBalance(ctx, alice.FormattedAddress(), "ustake")
+	require.NoError(t, err)
+	require.Equal(t, originalAmount, bal)
+
```

Comment:
same question about checking the usdc balance

Reply:
Added check in: https://github.com/noble-assets/noble/pull/373/commits/93925c17bb45c43e80cdce2516044b89fc9126ab